gensim/models/wrappers/fasttext.py

```diff
@@ -337,8 +367,26 @@ def init_ngrams(self):
         """
         self.wv.ngrams = {}
         all_ngrams = []
+        if bin_only:
+            self.wv.syn0 = zeros((len(self.wv.vocab), self.vector_size), dtype=REAL)
         for w, v in self.wv.vocab.items():
-            all_ngrams += self.compute_ngrams(w, self.wv.min_n, self.wv.max_n)
+            word_ngrams = self.compute_ngrams(w, self.wv.min_n, self.wv.max_n)
+            all_ngrams += word_ngrams
+
+            
+            if bin_only:
+                #self.wv.syn0 = zeros((len(self.wv.vocab), self.vector_size), dtype=REAL)
+                word_vec = np.zeros(self.wv.syn0.shape[1])
+
+                num_word_ngram_vectors = len(word_ngrams)
+                for word_ngram in word_ngrams:
```

Comment:
This seems redundant and possibly error prone.
Instead you could simply use `self.word_vec(word)` to obtain the vector for the in-vocab words, after the `syn0_all` trimming takes place. Could you please try that, and see if it initializes the correct values for `self.syn0`?